ibm_cloud_sdk_core/authenticators/bearer_authenticator.py

```diff
@@ -0,0 +1,59 @@
+# coding: utf-8
+
+# Copyright 2019 IBM All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from .authenticator import Authenticator
+from ..utils import has_bad_first_or_last_char
+
+
+class BearerAuthenticator(Authenticator):
+    authentication_type = 'bearerToken'
+
+    def __init__(self, bearer_token):
+        """
+        :attr str bearer_token: User managed bearer token
+        """
+        self.bearer_token = bearer_token
+        self.validate()
+
+    def validate(self):
+        """
+        Performs validation on input params
+        """
+        if self.bearer_token is None:
+            raise ValueError('The bearer token shouldn\'t be None.')
+
+        if has_bad_first_or_last_char(self.bearer_token):
+            raise ValueError(
+                'The bearer token shouldn\'t start or end with curly brackets or quotes. '
+                'Please remove any surrounding {, }, or \" characters.')
```

Comment:
This wasn't explicit, but I don't think we need to check the first and last character on a beaker token. I don't know that they would ever have those characters, but because they are auto-generated I feel weird about checking for something that could be there and be correct.

I could be wrong about that though

Reply:
I think you can also get the token from the cp4d console - maybe we should be checking for bad characters here? I don't remember what getting the credentials looks like in the console.

Reply:
Yeah maybe we should look at examples but it's still a generated string. Probably will never be a bracket or quotation at the beginning or end so this is probably not a bad thing but since we can't control it, it might be risky. And I don't know that it adds a lot of value for the risk

Reply:
Yeah we shouldn't have to. Here's a screenshot of the screen where you get the token and URL in CP4D:

<img width="1680" alt="Screen Shot 2019-08-19 at 10 32 19 AM" src="https://user-images.githubusercontent.com/8710772/63274862-8aa87e00-c26e-11e9-9b91-20bf1a24dc80.png">

It's basically the same as in public where you just hit a copy button to grab the right values.

Reply:
My vote is to avoid perform those "HasBadCharacters..." checks.  Those seem to be a remnant of a by-gone era :)  Hopefully...